src/kernel/imfilter_opencl.jl

```diff
@@ -0,0 +1,95 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+const array_kernel(::OpenCLMethod, array) = array
+
+const view_kernel(::OpenCLMethod, array, I) = view(array, I)
+
+function imfilter_kernel(::OpenCLMethod, img, krn)
+  # retrieve basic info
+  N = ndims(img)
```

Comment:
You are not using N in the code. Can you please be extra careful with these changes? Also, the T variable is only used in two specific lines, we can just call ComplexF64 there directly instead of introducing a new variable T?

Reply:
I removed the line `N = ndims(img)`, but the T variable is called 6 times.